src/main/java/com/funkydonkies/w4v3/curve/CustomCurveMesh.java

```diff
@@ -67,17 +67,19 @@ public void setMeshPoints(final int i) {
 	 * @param i the index of the splinepoint
 	 */
 	public void getSegmentPoint(final int i) {
+		final int mulLength = 3, offset = 2; 
 		 
 		if (splinePoints[i].getY() < splinePoints[i + 1].getY()) {
 			final Vector3f vecNew = new Vector3f().add(splinePoints[i + 1]);
 			vecNew.setY(splinePoints[i].getY());
-			meshStructurePoints[splinePoints.length * 3 + 2 * i] = splinePoints[i];
-			meshStructurePoints[splinePoints.length * 3 + 2 * i + 1] = vecNew;		
+			meshStructurePoints[splinePoints.length * mulLength + offset * i] = splinePoints[i];
+			meshStructurePoints[splinePoints.length * mulLength + offset * i + 1] = vecNew;		
```

Comment:
looks very clean, only thing maybe add parenthesis on these kind of things to make it even more clear